optimum_benchmark/task_utils.py

```diff
@@ -69,6 +69,9 @@ def infer_library_from_model_name_or_path(model_name_or_path: str, revision: Opt
         model_info = huggingface_hub.model_info(model_name_or_path, revision=revision)
         inferred_library_name = getattr(model_info, "library_name", None)
 
+        if "gguf" in model_info.tags:
+            inferred_library_name = "llama_cpp"
+
```

Comment:
I think the inferred library is enough in this case, no need to make `llama_cpp` the default for repos that have a `gguf` tag (a lot of official repos have both `safetensors` and `gguf` weights), maybe if `inferred_library_name` is still `None` at the end we can check the tags for `gguf`, otherwise the default should stay `transformers` for backward compatibility.

Reply:
Good point. So I changed it and now defined the library inside the config instead of inferring it. I also modified the infer_model_type_from_model_name_or_path and infer_task_from_model_name_or_path so that you can supply a library if it isdefined. This prevents them from inferring a different library that the one submitted in the config.